packages/flutter/lib/src/material/time.dart

```diff
@@ -85,7 +85,10 @@ class TimeOfDay {
   /// This is a shortcut for [MaterialLocalizations.formatTimeOfDay].
   String format(BuildContext context) {
     final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-    return localizations.formatTimeOfDay(this);
+    return localizations.formatTimeOfDay(
+      this,
+      alwaysUse24HourFormat: MediaQuery.of(context).alwaysUse24HourFormat,
```

Comment:
if you cannot handle MediaQuery.of() returning null (which seems reasonable, many material widgets require this) then please add an assert to the build method that calls [debugCheckHasMediaQuery](https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/debugCheckHasMediaQuery.html). (See other widgets for examples of doing this.)